code/modules/mob/living/silicon/ai/ai.dm

```diff
@@ -633,41 +633,89 @@ var/list/ai_list = list()
+				personnel_list["[t.fields["name"]]: [t.fields["rank"]]"] = t.fields["image"]//Pull names, rank, and image.
+
+			if(personnel_list.len)
+				input = input("Select a crew member:") as null|anything in personnel_list
+				var/icon/character_icon = personnel_list[input]
+				if(character_icon)
+					qdel(holo_icon)//Clear old icon so we're not storing it in memory.
+					holo_icon = getHologramIcon(icon(character_icon))
+			else
+				alert("No suitable records found. Aborting.")
+
+		if("Animal")
+			var/list/icon_list = list(
+			"bear",
+			"carp",
+			"chicken",
+			"corgi",
+			"cow",
+			"crab",
+			"fox",
+			"goat",
+			"cat",
+			"cat2",
+			"poly",
+			"pug",
+			"spider"
+			)
+
+			input = input("Please select a hologram:") as null|anything in icon_list
+			if(input)
+				qdel(holo_icon)
+				switch(input)
+					if("bear")
+						holo_icon = getHologramIcon(icon('icons/mob/animal.dmi', input))
+					if("carp")
+						holo_icon = getHologramIcon(icon('icons/mob/animal.dmi', input))
+					if("chicken")
+						holo_icon = getHologramIcon(icon('icons/mob/animal.dmi',"chicken_brown"))
+					if("corgi")
+						holo_icon = getHologramIcon(icon('icons/mob/pets.dmi', input))
+					if("cow")
+						holo_icon = getHologramIcon(icon('icons/mob/animal.dmi', input))
+					if("crab")
+						holo_icon = getHologramIcon(icon('icons/mob/animal.dmi', input))
+					if("fox")
+						holo_icon = getHologramIcon(icon('icons/mob/pets.dmi', input))
+					if("goat")
+						holo_icon = getHologramIcon(icon('icons/mob/animal.dmi', input))
+					if("cat")
+						holo_icon = getHologramIcon(icon('icons/mob/pets.dmi', input))
+					if("cat2")
+						holo_icon = getHologramIcon(icon('icons/mob/pets.dmi', input))
+					if("poly")
+						holo_icon = getHologramIcon(icon('icons/mob/animal.dmi',"parrot_fly"))
+					if("pug")
+						holo_icon = getHologramIcon(icon('icons/mob/pets.dmi', input))
+					if("spider")
+						holo_icon = getHologramIcon(icon('icons/mob/animal.dmi',"guard"))
 		else
-			alert("No suitable records found. Aborting.")
-
-	else
-		var/icon_list[] = list(
-		"default",
-		"floating face",
-		"xeno queen",
-		"space carp"
-		)
-		input = input("Please select a hologram:") as null|anything in icon_list
-		if(input)
-			qdel(holo_icon)
-			switch(input)
-				if("default")
-					holo_icon = getHologramIcon(icon('icons/mob/AI.dmi',"holo1"))
-				if("floating face")
-					holo_icon = getHologramIcon(icon('icons/mob/AI.dmi',"holo2"))
-				if("xeno queen")
-					holo_icon = getHologramIcon(icon('icons/mob/AI.dmi',"holo3"))
-				if("space carp")
-					holo_icon = getHologramIcon(icon('icons/mob/AI.dmi',"holo4"))
+			var/list/icon_list = list(
+			"default",
+			"floating face",
+			"xeno queen",
+			"horror"
+			)
+
+			input = input("Please select a hologram:") as null|anything in icon_list
+			if(input)
+				qdel(holo_icon)
+				switch(input)
+					if("default")
+						holo_icon = getHologramIcon(icon('icons/mob/AI.dmi', input))

```

Comment:
as you can see, all 4 cases are now identical, so the switch isn't needed at all (which was my initial point)